src/drivers/mmc/host/sdhci/sdhci.c

```diff
@@ -6,301 +6,30 @@
-
-	sdhci_writel(host, USDHC_INT_STATUS, -1);
-
-	sdhci_writel(host, USDHC_DS_ADDR, (uint32_t) buffer);
-
-	imx6_usdhc_send_cmd(host, 24, blkno * BLK_LEN, 0);
-
-	return count;
-}
-
-struct block_dev_driver imx6_usdhc_driver = {
-	.name  = "imx6_usdhc_driver",
-	.probe = NULL,
-	.read  = imx6_usdhc_read,
-	.write = imx6_usdhc_write
-};
-
-PERIPH_MEMORY_DEFINE(imx6_usdhc, BASE_ADDR, 0x4000 * MAX_HC_NUM);
-
-#if !NEW_API
-static uint32_t sdhc_get_size(struct sdhci_host *host) {
-	uint32_t t;
-	/* Assume it's not high-capacity SD card */
-	imx6_usdhc_send_cmd(host, 9, 0x4567 << 16, 7);
-	t = (sdhci_readl(host, USDHC_CMD_RSP1) >> 6) & 0x3;
-	t |= (sdhci_readl(host, USDHC_CMD_RSP1) >> 24) << 2;
-	t |= (sdhci_readl(host, USDHC_CMD_RSP2) & 0x3) << 10;
-
-	if (t == 0) {
-		return 0;
-	}
-	return 256 * 1024 * (t + 1);
-}
-#endif
-
-static inline void sdhci_mmc_request(struct mmc_host *mmc_host, struct mmc_request *req) {
-	struct sdhci_host *host;
-
-	assert(mmc_host);
-	assert(req);
-
-	host = mmc_host->priv;
-
-	log_debug("req->cmd.opcode %x, req->cmd.arg %x, req->cmd.flags %x", req->cmd.opcode, req->cmd.arg, req->cmd.flags);
-
-	if (req->cmd.flags & MMC_DATA_XFER) {
-		sdhci_writel(host, USDHC_INT_STATUS, -1);
-
-		sdhci_writel(host, USDHC_DS_ADDR, (uint32_t) req->data.addr);
-	}
-
-	imx6_usdhc_send_cmd(host, req->cmd.opcode, req->cmd.arg, req->cmd.flags);
-
-	sdhci_cmd_done(host, req);
-}
-
-static const struct mmc_host_ops sdhci_mmc_ops = {
-	.request = sdhci_mmc_request,
-};
-
 static struct sdhci_host sdhci_hosts[MAX_HC_NUM];
 
-static  inline int sdhci_hw_probe(struct sdhci_host *host) {
-	uint32_t tmp;
-#if NEW_API
-	struct mmc_host *mmc;
-#else
-	struct block_dev *bdev;
-	size_t bsize;
-#endif
-	/* SDHC initialization */
-	imx6_usdhc_reset(host);
-
-	sdhci_writel(host, USDHC_SYS_CTRL, 0xb011f);
-	sdhci_writel(host, USDHC_INT_STATUS_EN, -1);
-	sdhci_writel(host, USDHC_PROT_CTRL, sdhci_readl(host, USDHC_PROT_CTRL) | 1 | 4);
-
-	tmp = 512 | (1 << 16);
-	sdhci_writel(host, USDHC_BLK_ATT, tmp);
-//	imx6_usdhc_set_block_len(host, BLK_LEN);
-#if !NEW_API
-	/* SD initialization */
-	imx6_usdhc_send_cmd(host, 0, 0, 0);
-	imx6_usdhc_send_cmd(host, 8, 0x1AA, 0);
-	imx6_usdhc_send_cmd(host, 55, 0x0, 0x15);
-	imx6_usdhc_send_cmd(host, 41, 0x4000FFFF, 1);
-
-	imx6_usdhc_send_cmd(host, 2, 0, 7);
-	imx6_usdhc_send_cmd(host, 3, 0, 15);
-
-	bsize = sdhc_get_size(host);
-	log_debug("bdev size is %d\n", bsize);
-	if (bsize == 0) {
-		return 0;
-	}
-	imx6_usdhc_send_cmd(host, 7, 0x4567 << 16, 0);
-
-	bdev = block_dev_create("sdhc", &imx6_usdhc_driver, host);
-
-	bdev->block_size = BLK_LEN;
-	bdev->size = bsize;
-#else /* !NEW_API */
-	if (sdhci_readl(host, USDHC_PRES_STATE) & USDHC_PRES_STATE_CDPL) {
-		mmc = mmc_alloc_host();
-		mmc->ops = &sdhci_mmc_ops;
-		mmc->priv = host;
-		mmc_sw_reset(mmc);
-	}
-#endif /* !NEW_API */
-
-	_reg_dump(host);
-
-	return 0;
-}
+PERIPH_MEMORY_DEFINE(imx6_usdhc, BASE_ADDR, 0x4000 * MAX_HC_NUM);
```

Comment:
May be use macro
`#define IO_REG_SIZE 0x4000`

Reply:
Done